src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs

```diff
@@ -53,6 +53,11 @@ public void Build(FieldType contentQuery, ContentTypeDefinition contentTypeDefin
                     continue;
                 }
 
+                if (!(part.PartDefinition.Fields.Any(field=> contentFieldProviders.Any(fieldProvider=>fieldProvider.GetField(field)!=null))))
```

Comment:
This line will be more readable if you use a regular foreach loop. Also, fix the spacing

Reply:
It's not really a loop. It's a boolean expression testing that no standard field types are included in the current ContentPart.

Reply:
well, Any loop over the records.

Reply:
That would be duplicating the logic already in the fieldProvider and, IMO, it would make the process a lot more complicated that it is right now.

That being said, if the Orchard Core maintainers think it would improve the project, their welcome to create all that extra logic. This commit fixes the issue so I'm good with it as is.